agenta-web/src/components/Playground/NewVariantModal.tsx

```diff
@@ -30,22 +30,33 @@ const NewVariantModal: React.FC<Props> = ({
 }) => {
     const classes = useStyles()
     const [variantPlaceHolder, setVariantPlaceHolder] = useState("Source Variant")
+    const [isInputValid, setIsInputValid] = useState(false) // To check if the input is valid
+
     const handleTemplateVariantChange = (value: string) => {
         let newValue = value.includes(".") ? value.split(".")[0] : value
         setTemplateVariantName(value)
         setVariantPlaceHolder(`${newValue}`)
     }
 
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const inputValue = e.target.value
+        setNewVariantName(inputValue)
+        setIsInputValid(inputValue.trim().length > 0)  // Update the validity of the input
+    }
+
     return (
         <Modal
             title="Create a New Variant"
             open={isModalOpen}
             onOk={() => {
-                setIsModalOpen(false)
-                addTab()
+                if (isInputValid) { // Check if the input is valid
```

Comment:
please prefer the early return pattern

Reply:
@seungduk-yanolja thank you for double checking! Will keep it in mind.